### PR TITLE
Transactional cancellation, sync fixes

### PR DESCRIPTION
**Pull request by hdevalence**
This PR makes a number of improvements to the sync algorithm, primarily by trying to make sure that all of the expectations of the components of the verification pipeline are aligned.

It also contains improved diagnostics, and better defaults that avoid block download storms (by lowering the maximum number of concurrent block requests).

### CODE REVIEW

**Comment by hdevalence** [comment]
Rebased onto `main` and resolved conflicts.

File: zebrad/src/components/sync.rs [hunk 1]
```diff
@@ -214,9 +214,14 @@ where
...
+
                         Err(e) => {
-                            tracing::warn!(?e, "error downloading and verifying block");
-                            continue 'sync;
+                            if format!("{:?}", e).contains("AlreadyVerified {") {
+                                tracing::info!(error = ?e, "error seems like it might be already verified, this is probably fine i guess?")
+                            } else {
+                                tracing::warn!(?e, "error downloading and verifying block");
+                                continue 'sync;
+                            }
```
Comment (hdevalence) [inline zebrad/src/components/sync.rs:224]: We should replace this with a better method of error-type detection but I don't think it's essential to do so until after the alpha.

Comment (yaahc) [reply zebrad/src/components/sync.rs:224]: sounds good

File: zebra-state/src/sled_state.rs [hunk 2]
```diff
@@ -381,3 +380,62 @@ impl FinalizedState {
...
+        ?height,
+        transaction_count,
+        transparent_prevout_count,
+        transparent_newout_count,
+        sprout_nullifier_count,
+        sapling_nullifier_count,
+        "preparing to commit finalized block"
+    );
+    metrics::counter!(
+        "state.sled.committed.transactions",
```
Comment (hdevalence) [inline zebra-state/src/sled_state.rs]: We might want a different name for these metrics keys.

Comment (hdevalence) [reply zebra-state/src/sled_state.rs]: Changed in a fixup commit that I'll squash before merging.

**Review by yaahc — DISMISSED** [review]

**Review by yaahc — APPROVED** [review]

**Pull request merged**